core/symbol.rbs

```diff
@@ -176,7 +174,7 @@ class Symbol
   # variables; see String#=~.
   #
   def =~: (Regexp regex) -> Integer?
-        | [T] (String::_MatchAgainst[self, T] object) -> T
+        | [T] (String::_MatchAgainst[String, T] object) -> T
```

Comment:
[Optional] Wondering if you can add a test to confirm the `_MatchAgains` generic interface works correctly with Steep. (in `test/typecheck/_MatchAgainst` or somewhere.)

Reply:
Do we have any of these as examples to base it off of?